test/models/ride_test.dart

```diff
@@ -0,0 +1,468 @@
+                )
+                .toJsonForApi()
+          ]));
+      expect(
+          await Ride.userHasRideAtTimeRange(
+              DateTimeRange(start: DateTime.now(), end: DateTime.now().add(const Duration(hours: 10))), 2),
+          true);
+    });
+    test('has no ride at time range', () async {
+      when.call(rideProcessor.processUrl(any)).thenReturn(jsonEncode([
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  status: RideStatus.approved,
+                  startTime: DateTime.now().add(const Duration(hours: 8)),
+                  endTime: DateTime.now().add(const Duration(hours: 10)),
+                )
+                .toJsonForApi(),
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 2)),
+                  endTime: DateTime.now().add(const Duration(hours: 4)),
+                )
+                .toJsonForApi(),
+          ]));
+      expect(
+          await Ride.userHasRideAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 4)),
+                end: DateTime.now().add(const Duration(hours: 6)),
+              ),
+              2),
+          false);
+    });
+    test('not approved ride overlaps with time range', () async {
+      when.call(rideProcessor.processUrl(any)).thenReturn(jsonEncode([
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  status: RideStatus.pending,
+                  startTime: DateTime.now().add(const Duration(hours: 7)),
+                  endTime: DateTime.now().add(const Duration(hours: 10)),
+                )
+                .toJsonForApi(),
+          ]));
+      expect(
+          await Ride.userHasRideAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 6)),
+                end: DateTime.now().add(const Duration(hours: 8)),
+              ),
+              2),
+          false);
+    });
+    test('ride overlaps with time range', () async {
+      when.call(rideProcessor.processUrl(any)).thenReturn(jsonEncode([
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  status: RideStatus.approved,
+                  startTime: DateTime.now().add(const Duration(hours: 7)),
+                  endTime: DateTime.now().add(const Duration(hours: 10)),
+                )
+                .toJsonForApi(),
+          ]));
+      expect(
+          await Ride.userHasRideAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 6)),
+                end: DateTime.now().add(const Duration(hours: 8)),
+              ),
+              2),
+          true);
+    });
+  });
+  group('Ride.shouldShowInListView', () {
+    test('show in Listview', () {
```

Comment:
```suggestion
    test('show in ListView', () {
```
Also, isn't this already covered by the trip test?

Reply:
here are new cases since it is overwriten in ride. In drive it is not overwriten so it is not tested in drive